chemprop/data/datasets.py

```diff
@@ -355,9 +355,9 @@ def normalize_targets(self, scaler: StandardScaler | None = None) -> StandardSca
         return self.datasets[0].normalize_targets(scaler)
 
     def normalize_inputs(
-        self, key: str | None = "X_d", scaler: StandardScaler | None = None
+        self, key: str | None = "X_d", scaler: list[StandardScaler] | None = None
     ) -> list[StandardScaler]:
-        return [dset.normalize_inputs(key, scaler) for dset in self.datasets]
+        return [dset.normalize_inputs(key, scaler) for dset, scaler in zip(self.datasets, scaler)] if scaler is not None else [dset.normalize_inputs(key) for dset in self.datasets]
```

Comment:
Also X_d is only needed once per multicomponent datapoint. So there only needs to be one X_d_scaler put dataset regardless of whether it is mono or multicomponent. Currently this will make a list of identical X_d_scaler's because we save X_d in every dataset. That will work but seems a bit wasteful for dataspace. Maybe we should change the CLI code to only save X_d once in a multicomponent datapoint, but that should probably be in a different PR.

Reply:
Oooof this is a good catch. We should definitely add a test for additional features for mol+rxn. I will work on this.

Reply:
Yeah I think changing the CLI to save X_d once should be in another PR. In the same PR, we should also change it such that the targets are only saved in the first component. Currently targets are saved in all components, even though only the copy in the first componenets are being used.